### ISSUE TITLE
Is there a flag to *statically* link SQLCipher?

### ISSUE THREAD

**Issue by aguynamedben** [mapbox/node-sqlite3#1065]
Hi, I'm building an Electron app with SQLCipher. Following the README instructions for SQLCipher, everything worked in development, but when I deployed the new version of my app, it failed because SQLCipher was dynamically linked and my users haven't installed SQLCipher on their computer.

Resources:
* [Original issue that led to SQLCipher support](https://github.com/mapbox/node-sqlite3/issues/195)
* [node-sqlite3's binding.gyp](https://github.com/mapbox/node-sqlite3/blob/master/binding.gyp)
* [How to tweak binding.gyp to statically link instead of dynamically link](https://github.com/nodejs/node-gyp/issues/328#issuecomment-24335081)
* [Maybe some other useful info on static vs. dynamic linking via gcc flags](https://stackoverflow.com/a/938947/3516664)

**Comment by springmeyer** [comment]
Yes, it is reasonable to add an option for this. The link to the static library will need to be completed to be absolute to work right.

**Comment by aguynamedben** [comment]
@springmeyer Thanks for the feedback.

I'm still testing this within the context of my app, but I submitted a PR that allows `--sqlite-static-library=/full/path/to/libsqlcipher.a` to be defined. Let me know if something looks off in there or you prefer another way of passing the static library path.

**Comment by aguynamedben** [comment]
@springmeyer My final effort toward making static linking for Electron+SQLCipher users is here: https://github.com/mapbox/node-sqlite3/pull/1069 It's a docs-only PR that explains the problem.

Providing docs to explain this issue to Electron+SQLCipher users seems like a better solution than to further complicate logic in binding.gyp. Plus I'm a n00b when it comes to gyp, compiler flags, and linkers.

This is an issue isolated to Electron+SQLCipher users, and it's simple to overwrite ./node_modules/sqlite3/binding.gyp with a modified version, rebuild, and package.
